src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,124 @@
+//Alvin Bautista
+//CS 410 - JUNIT Assignment
+//06112023
+
 public class StringManipulation implements StringManipulationInterface {
 
+    private String string; //stores the string to manipulate
+
     @Override
     public String getString() {
-        return null;
+        return this.string; //return the string stored in this object
     }
 
     @Override
     public void setString(String string) {
+        this.string = string; //set the string stored in this object
     }
 
     @Override
     public int count() {
-        return 0;
+        //check if the string is null or empty
+        if (this.string == null || this.string.trim().isEmpty()) {
+            return 0;
+        }
+        //split the string into words and return the number of words
+        return this.string.trim().split("\\s+").length;
```

Comment:
Alvin, this is a really concise line of code! I can see this method being easily overcomplicated, but you found a very elegant solution to this method. Your use of regex in the return statement was a smart idea that led you to avoid initializing an array to count the words in the string. 

I'm curious, though, if this method were to be implemented in industry, do you think SWEs would prefer to explicitly initialize an empty array first then fill it with strings before returning the length of the array? In other words, what is the general stance on the readability of concatenating methods on top of one-another?

Reply:
@mdbonner 

It is probably not preferred industry in production code. For one, it is concise but is not as readable. It would also be good to be able to see intermediate steps so bugs are more traceable. 

Although for our purposes, clear and concise code is easier for practicing code reviews.

Reply:
I agree with you on the readability of code, especially for junior SWEs reviewing code. This reminds of the concept of "code zen" that was taught in CS 210/211, good times.